lib/ash_ai.ex

```diff
@@ -243,10 +315,14 @@ defmodule AshAi do
     lang_chain
     |> LLMChain.add_tools(tools)
     |> then(fn llm_chain ->
-      if opts.actor do
+      if opts.actor || opts.on_tool_start || opts.on_tool_end do
```

Comment:
lets just remove this conditional. Looking at it, it already has a bug where it only sets the tenant if the actor is present 😓

Reply:
we can just always set this context.